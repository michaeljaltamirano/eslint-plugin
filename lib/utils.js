"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isObjectLike = exports.isNotNullish = exports.isJson = void 0;
function isNotNullish(arg) {
    return arg !== null && arg !== undefined;
}
exports.isNotNullish = isNotNullish;
function isObjectLike(arg) {
    return (typeof arg === 'object' && isNotNullish(arg) && !!Object.keys(arg).length);
}
exports.isObjectLike = isObjectLike;
function isJson(arg) {
    if (!isNotNullish(arg))
        return false;
    return isObjectLike(arg);
}
exports.isJson = isJson;
