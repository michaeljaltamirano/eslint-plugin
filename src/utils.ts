function isNotNullish<T>(arg: null | undefined | T): arg is T {
	return arg !== null && arg !== undefined;
}

function isObjectLike(arg: unknown): arg is Record<string, unknown> {
	return (
		typeof arg === 'object' && isNotNullish(arg) && !!Object.keys(arg).length
	);
}

function isJson(
	arg: unknown | Record<string, unknown>,
): arg is Record<string, unknown> {
	if (!isNotNullish(arg)) return false;

	return isObjectLike(arg);
}

export { isJson, isNotNullish, isObjectLike };
