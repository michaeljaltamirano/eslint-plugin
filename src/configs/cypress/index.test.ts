import { cypress } from '.';
import { generateTest } from '../../tests/utils';

describe('@michaeljaltamirano/eslint-plugin', () => {
	describe('cypress config', () => {
		generateTest(__dirname, cypress);
	});
});
