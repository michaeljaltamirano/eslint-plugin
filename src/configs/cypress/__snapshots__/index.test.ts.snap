// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@michaeljaltamirano/eslint-plugin cypress config serializes to a snapshot 1`] = `
Object {
  "env": Object {
    "cypress/globals": true,
  },
  "globals": Object {},
  "ignorePatterns": Array [],
  "parserOptions": Object {},
  "plugins": Array [
    "cypress",
  ],
  "rules": Object {
    "cypress/assertion-before-screenshot": Array [
      "error",
    ],
    "cypress/no-assigning-return-values": Array [
      "error",
    ],
    "cypress/no-async-tests": Array [
      "error",
    ],
    "cypress/no-force": Array [
      "error",
    ],
    "cypress/no-pause": Array [
      "error",
    ],
    "cypress/no-unnecessary-waiting": Array [
      "error",
    ],
    "cypress/require-data-selectors": Array [
      "off",
    ],
  },
  "settings": Object {},
}
`;
